mitmproxy/flow/export.py

```diff
@@ -4,23 +4,34 @@
 import re
 from textwrap import dedent
 
-from six.moves.urllib.parse import quote, quote_plus
+import six
+from six.moves import urllib
 
 import netlib.http
 
 
+def _native(s):
+    if six.PY2:
+        if isinstance(s, six.text_type):
+            return s.encode()
+    else:
+        if isinstance(s, six.binary_type):
+            return s.decode()
+    return s
+
+
 def dictstr(items, indent):
     lines = []
     for k, v in items:
-        lines.append(indent + "%s: %s,\n" % (repr(k), repr(v)))
+        lines.append(indent + "%s: %s,\n" % (repr(_native(k)), repr(_native(v))))
     return "{\n%s}\n" % "".join(lines)
 
 
 def curl_command(flow):
     data = "curl "
 
     for k, v in flow.request.headers.fields:
```

Comment:
You can use
```python
for k, v in flow.request.headers.items():
```

It's not 100% exact as headers are folded, but if we follow the RFCs that does not change anything.